dotnet/src/Connectors/Connectors.AzureOpenAI/Core/AzureClientCore.ChatCompletion.cs

```diff
@@ -31,19 +33,36 @@ protected override OpenAIPromptExecutionSettings GetSpecializedExecutionSettings
     protected override Dictionary<string, object?> GetChatCompletionMetadata(OpenAIChatCompletion completions)
     {
 #pragma warning disable AOAI001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
-        return new Dictionary<string, object?>
+        var metadata = new Dictionary<string, object?>
         {
             { nameof(completions.Id), completions.Id },
             { nameof(completions.CreatedAt), completions.CreatedAt },
-            { ContentFilterResultForPromptKey, completions.GetContentFilterResultForPrompt() },
             { nameof(completions.SystemFingerprint), completions.SystemFingerprint },
             { nameof(completions.Usage), completions.Usage },
-            { ContentFilterResultForResponseKey, completions.GetContentFilterResultForResponse() },
-
             // Serialization of this struct behaves as an empty object {}, need to cast to string to avoid it.
             { nameof(completions.FinishReason), completions.FinishReason.ToString() },
             { nameof(completions.ContentTokenLogProbabilities), completions.ContentTokenLogProbabilities },
         };
+
+        try
+        {
+            metadata[ContentFilterResultForPromptKey] = completions.GetContentFilterResultForPrompt();
+            metadata[ContentFilterResultForResponseKey] = completions.GetContentFilterResultForResponse();
+        }
+        catch (MissingFieldException ex)
+        {
+            if (!ex.Message.Contains("OpenAI.Chat.ChatCompletion._serializedAdditionalRawData"))
+            {
+                throw;
+            }
+
+            if (this.Logger!.IsEnabled(LogLevel.Error))
+            {
+                this.Logger.LogError(ex, "Unsupported OpenAI SDK version for the current Azure OpenAI SDK version");
```

Comment:
If we reach this point it means we cannot add some of the metadata but otherwise everything works, is that correct?

Can we downgrade to a warning here instead?

Reply:
Im going to remove the try catch block, I think with the latest being stable we dont need to outwork this scenario, and recommend customers to update to latest instead.